src/Patient1.java

```diff
@@ -108,21 +126,22 @@ public String toString() {
 
```

Comment:
I noticed the common methods above. These are typically implemented at the abstract class level, so the behavior is consistent across implementations.

Reply:
Oh, I'm realizing this is just a separate component. You might consider replacing this file with a record:
```java
public record Patient(String name, String pcp, ArrayList<String> dates) {]